server.js

```
@@ -0,0 +1,221 @@
+        console.error(error);
+    })
+    req.on('end', () => {
+        let urlType;
+        let URL = req.url;
+        // GET METHOD
+        if (req.method === 'GET') {
+            if (req.url === '/' || req.url === '') {
+                urlType = 'html';
+                URL = '/index.html';
+            } else {
+                urlType = req.url.split('.');
+                urlType = urlType[1];
+            }
+            fs.readdir(`./public/`, (err, files) => {
+                if (files.includes(URL.slice(1)) === false) {
+                    URL = `/404.html`;
+                    urlType = 'html';
+                }
+                fs.readFile(`./public${URL}`, (err, data) => {
+                    if (err) {
+                        return console.log('Could not write the file.');
+                    }
+                    res.writeHead(200, {
+                        'Date': `${date}`,
+                        'Content-Type': `text/${urlType}`,
+                        'Content-Length': data.toString().length,
+                    })
+                    res.write(data.toString());
+                    res.end();
+                })
+            })
+
+        }
+        // POST METHOD
+        if (req.method === 'POST') {
+            if (req.url === '/elements') {
+                let parsedInfo = querystring.parse(body);
+                const template = `<!DOCTYPE html>
+<html lang="en">
+<head>
+<meta charset="UTF-8">
+<title>The Elements - ${parsedInfo.elementName}</title>
+<link rel="stylesheet" href="/styles.css">
+</head>
+<body>
+    <h1>${parsedInfo.elementName}</h1>
+    <h2>${parsedInfo.elementSymbol}</h2>
+    <h3>${parsedInfo.elementAtomicNumber}</h3>
+    <p>${parsedInfo.elementDescription}</p>
+    <p><a href="/">back</a></p>
+</body>
+</html>`;
+                fs.writeFile(`./public/${parsedInfo.elementName}.html`, template, (err) => {
+                    if (err) {
+                        return console.log(err);
```

This should return an error to the client using either a 400/500 error